docs/source/format/Columnar.rst

```diff
@@ -1078,8 +1078,95 @@ Arrow columnar format. Each logical type uses one of the above
 physical layouts. Nested logical types may have different physical
 layouts depending on the particular realization of the type.
 
-We do not go into detail about the logical types definitions in this
-document as we consider `Schema.fbs`_ to be authoritative.
+.. glossary::
+
+   Null Type
+      Array where all values are null.
+
+   Boolean Type
+      Array of bit-packed values.
+
+   Int Type
+      Array of signed or unsigned integers with 8, 16, 32 or 64 bit width.
+
+   Floating Point Type
+      Array of floating point numbers with half, single or double precision.
+
+   .. TODO: list other types if we agree to keep the glossary
+
+
+Types and their respective Physical Memory Layout
+-------------------------------------------------
+
++--------------------+----------------------------------+------------------------------+
+| Type               | Physical Memory Layout           | Additional info stored       |
```

Comment:
I'm not sure what "additional info" alludes to? This seems to be mostly type parameters, but not only. Mixing concepts together like this feels confusing to me.

Reply:
I added all additional parameters from `Schema.fbs` and only removed some from `Dictionary`, I think, because those are used for IPC and not for the format.

Reply:
We could change "Additional info stored" to "Parameters"?